blog/2015-04-02-devstudio-8.1.0.GA-for-luna.adoc

```diff
@@ -0,0 +1,107 @@
+
+== Upgraded Forge 2 Runtime
+
+The Forge 2 runtime was upgraded to 2.15.2.Final. See the full release notes link:http://forge.jboss.org/news/2015-03-19/forge-2.15.2.final-%28morning-star%29-is-here[here.]
+
+There is also an improvement in Forge wizards which now render notes below the inputs (where available):
+
+image::/documentation/whatsnew/forge/images/4.2.3.CR1/note.png[]
+
+== HTML5 & JavaScript Editing Improvements
+
+HTML validation in Eclipse Web Tools Luna has an annoying issue with custom HTML5 tags. It warns about any HTML tag name that is not mentioned in the HTML specification. We contributed a patch to Eclipse Mars which allows to disable those warnings for any particular tag name (or a tag name mask, e.g. <ion-*>). Though this was fixed in Eclipse Mars only we back ported this useful feature to JBoss Tools 4.2.3 which is based on Eclipse Luna SR2:
+
+image::/documentation/whatsnew/jst/images/4.2.3.Beta1/validation.png[]
+
+JavaSript file validation now respects ECMA5 keywords.   
+
+== Tern.java and AngularJS Eclipse 0.8.2
+
+Tern.java and AngularJS Eclipse (available via JBoss Central - Early Access) has been upgraded to 0.8.2 version.
+
+There is a bunch of new features which were introduced in 0.8.0 releases: 
+
+- https://github.com/angelozerr/tern.java/wiki/New-and-Noteworthy-0.8.0[Tern.java 0.8.0 New and Noteworthy]
+- https://github.com/angelozerr/angularjs-eclipse/wiki/New-and-Noteworthy-0.8.0[AngularJS Eclipse Plugin 0.8.0 New and Noteworthy]
+
+0.8.2 releases also include bug fixes for some critical issues which caused Eclipse to freeze.
+
+== Multiple Browser Engine Support on Linux in Visual Editor 
+
+On Linux Visual Page Editor can be opened in HTML5 mode for HTML files, where all HTML5 features are supported.
+Visual Page editor mode can be changed with dialog, which appears during HTML file opening or in Visual Page Editor Preferences.
+
+image::/documentation/whatsnew/vpe/images/4.2.3.Beta1/engine_dialog.png[]
+
+== Deprecation of Visual Editor for JSF Pages
+
+Currently JBoss Tools and JBoss Developer Studio have some problems with supporting both HTML5 Visual Preview and JSF Visual Editor.
+It's related to the outdated XULRunner which is used in JBoss Tools to provide visual editing feature for JSF pages.
+It has very limited HTML5 support and conflicts with other browsers which may be used for HTML5 preview.    
+It's time now to deprecate the visual part of XHTML and JSP Page editor. JBoss Developer Studio 8.1 and JBoss Tools 4.2.3 still include JSF Visual Editor but it will be removed in future releases. 
+
+This deprecation is only about visual part of the XHTML/JSP Editor (Visual/Source and Preview tabs). We don't have any plans to remove functionality from the source editor.
```

Comment:
"We don't have any plans..." sounds like we might eventually do it...so I suggest instead to write  "The existing content assist, code navigation, wizards etc. related to JSF will still be present in the source editor."